contracts/mainnet/connectors/liquity/main.sol

```diff
@@ -0,0 +1,347 @@
+pragma solidity ^0.7.6;
+
+/**
+ * @title Liquity.
+ * @dev Lending & Borrowing.
+ */
+import "hardhat/console.sol";
+
+import { BorrowerOperationsLike, TroveManagerLike, StabilityPoolLike, StakingLike } from "./interface.sol";
+import { Stores } from "../../common/stores.sol";
+import { Helpers } from "./helpers.sol";
+import { Events } from "./events.sol";
+
+abstract contract LiquityResolver is Events, Helpers {
+    BorrowerOperationsLike internal constant borrowerOperations =
+        BorrowerOperationsLike(0x24179CD81c9e782A4096035f7eC97fB8B783e007);
+    TroveManagerLike internal constant troveManager =
+        TroveManagerLike(0xA39739EF8b0231DbFA0DcdA07d7e29faAbCf4bb2);
+    StabilityPoolLike internal constant stabilityPool =
+        StabilityPoolLike(0x66017D22b0f8556afDd19FC67041899Eb65a21bb);
+    StakingLike internal constant staking =
+        StakingLike(0x4f9Fbb3f1E99B56e0Fe2892e623Ed36A76Fc605d);
+
+    struct AdjustTrove {
+        uint maxFeePercentage;
+        uint withdrawAmount;
+        uint depositAmount;
+        uint borrowAmount;
+        uint repayAmount;
+        bool isBorrow;
+    }
+
+    constructor() {
+        console.log("Liquity Connector contract deployed at", address(this));
+    }
+
+    /* Begin: Trove */
+
+    /**
+     * @dev Deposit native ETH and borrow LUSD
+     * @notice Opens a Trove by depositing ETH and borrowing LUSD
+     * @param depositAmount The amount of ETH to deposit
+     * @param maxFeePercentage The maximum borrow fee that this transaction should permit 
+     * @param borrowAmount The amount of LUSD to borrow
+     * @param upperHint Address of the Trove near the upper bound of where the user's Trove will now sit in the ordered Trove list
+     * @param lowerHint Address of the Trove near the lower bound of where the user's Trove will now sit in the ordered Trove list
+     * @param getId Optional storage slot to retrieve ETH instead of receiving it from msg.value
+     * @param setId Optional storage slot to store the LUSD borrowed against
+    */
+    function open(
+        uint depositAmount,
+        uint maxFeePercentage,
+        uint borrowAmount,
+        address upperHint,
+        address lowerHint,
+        uint getId,
+        uint setId
+    ) external payable returns (string memory _eventName, bytes memory _eventParam) {
+        // User can either send ETH directly or have it collected from a previous spell
+        depositAmount = getUint(getId, depositAmount);
+
+        borrowerOperations.openTrove{value: depositAmount}(
+            maxFeePercentage,
+            borrowAmount,
+            upperHint,
+            lowerHint
+        );
+
+        // Allow other spells to use the borrowed amount
+        setUint(setId, borrowAmount);
```

Comment:
I guess the same “TODO” comment about the fee applies here.

Reply:
Borrow amount is the exact amount they will borrow, not the total debt, so I don't believe there's anything to change here